lib/spack/docs/packaging_guide.rst

```diff
@@ -2071,19 +2150,15 @@ The parameters are as follows:
     targets into.  It acts like a string, but it's actually its own
     special type, :py:class:`Prefix <spack.util.prefix.Prefix>`.
 
-``spec`` and ``prefix`` are passed to ``install`` for convenience.
-``spec`` is also available as an attribute on the package
-(``self.spec``), and ``prefix`` is actually an attribute of ``spec``
-(``spec.prefix``).
+The arguments ``spec`` and ``prefix`` are passed only for convenience, as they always
+correspond to ``self.spec`` and ``self.spec.prefix`` respectively.
```

Comment:
@alalazo: before I get carried away with dependency injection, though, it might be useful to look at how this would affect the spec attributes API.  *Those* methods have some mandated parameters/structure, so they're a little different.